Importing Dependencies
--

In [1]:
import numpy as np
import pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn import svm
from sklearn.metrics import accuracy_score

Data Collection and Analysis
--

In [2]:
#Loading the diabetes dataset to pandas dataframe
diabetes_dataset = pd.read_csv(r"C:\Users\VIVAH\Desktop\OneCampus\Database\diabetes.csv")

#Printing the first five rows of the diabetes dataset
diabetes_dataset.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [3]:
# getting the numbers of Rows and Columns
diabetes_dataset.shape


(768, 9)

In [4]:
# getting the statistical measures of the data
diabetes_dataset.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [5]:
# how many cases are there for diabetes cases and none diabetes cases
diabetes_dataset['Outcome'].value_counts()

0    500
1    268
Name: Outcome, dtype: int64

0 ==> showing 500 patients are without diabetes
--
1 ==> showing 268 patients are with diabetes
--

In [42]:
diabetes_dataset.groupby('Outcome').mean()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
Outcome                                                                      
0           3.298000  109.980000      68.184000      19.664000   68.792000   
1           4.865672  141.257463      70.824627      22.164179  100.335821   

               BMI  DiabetesPedigreeFunction        Age  
Outcome                                                  
0        30.304200                  0.429734  31.190000  
1        35.142537                  0.550500  37.067164

In [9]:
X = diabetes_dataset.drop(columns='Outcome', axis=1)
Y = diabetes_dataset['Outcome']

print(X.head(),Y.head())

   Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  
0                     0.627   50  
1                     0.351   31  
2                     0.672   32  
3                     0.167   21  
4                     2.288   33   0    1
1    0
2    1
3    0
4    1
Name: Outcome, dtype: int64


Standardizing Our Data
--

In [10]:
scaler = StandardScaler()

In [11]:
# this fits and transforms the data in a common range (0 to 1)
standardized_data = scaler.fit_transform(X)

In [12]:
standardized_data

array([[ 0.63994726,  0.84832379,  0.14964075, ...,  0.20401277,
         0.46849198,  1.4259954 ],
       [-0.84488505, -1.12339636, -0.16054575, ..., -0.68442195,
        -0.36506078, -0.19067191],
       [ 1.23388019,  1.94372388, -0.26394125, ..., -1.10325546,
         0.60439732, -0.10558415],
       ...,
       [ 0.3429808 ,  0.00330087,  0.14964075, ..., -0.73518964,
        -0.68519336, -0.27575966],
       [-0.84488505,  0.1597866 , -0.47073225, ..., -0.24020459,
        -0.37110101,  1.17073215],
       [-0.84488505, -0.8730192 ,  0.04624525, ..., -0.20212881,
        -0.47378505, -0.87137393]])

In [13]:
#Reassign the values

X = standardized_data
Y = diabetes_dataset['Outcome']

print(X,Y)

[[ 0.63994726  0.84832379  0.14964075 ...  0.20401277  0.46849198
   1.4259954 ]
 [-0.84488505 -1.12339636 -0.16054575 ... -0.68442195 -0.36506078
  -0.19067191]
 [ 1.23388019  1.94372388 -0.26394125 ... -1.10325546  0.60439732
  -0.10558415]
 ...
 [ 0.3429808   0.00330087  0.14964075 ... -0.73518964 -0.68519336
  -0.27575966]
 [-0.84488505  0.1597866  -0.47073225 ... -0.24020459 -0.37110101
   1.17073215]
 [-0.84488505 -0.8730192   0.04624525 ... -0.20212881 -0.47378505
  -0.87137393]] 0      1
1      0
2      1
3      0
4      1
      ..
763    0
764    0
765    0
766    1
767    0
Name: Outcome, Length: 768, dtype: int64


Train Test Spilt

In [18]:
X_train, X_test, Y_train, Y_test = train_test_split(X,Y, test_size=0.2, stratify=Y, random_state=2)

In [19]:
print(X.shape, X_train.shape, X_test.shape)

(768, 8) (614, 8) (154, 8)


Training the Model
--

In [22]:
classifier = svm.SVC(kernel='linear')

In [23]:
#training our support vector Machine
classifier.fit(X_train, Y_train)

SVC(kernel='linear')

Model Evaluation 
--
Accuracy Score
--

In [26]:
#Accuracy score onthe training data
X_train_predict = classifier.predict(X_train) 
training_data_accuracy = accuracy_score(X_train_predict, Y_train)

In [27]:
print('Accuracy score of the trained data : ', training_data_accuracy)

Accuracy score of the trained data :  0.7866449511400652


In [28]:
#Accuracy score onthe test data
X_test_predict = classifier.predict(X_test) 
test_data_accuracy = accuracy_score(X_test_predict, Y_test)

In [29]:
print('Accuracy score of the test data : ', test_data_accuracy)

Accuracy score of the test data :  0.7727272727272727


Making a Predictive Model
--

In [30]:
input_data = (4,110,92,0,0,37.6,0.191,30)
#Convert to a numpy array
data = np.asarray(input_data)

In [31]:
data

array([  4.   , 110.   ,  92.   ,   0.   ,   0.   ,  37.6  ,   0.191,
        30.   ])

In [32]:
#reshaping array as we predicting one instance
data_reshaped = data.reshape(1,-1)

In [34]:
data_reshaped

array([[  4.   , 110.   ,  92.   ,   0.   ,   0.   ,  37.6  ,   0.191,
         30.   ]])

In [35]:
#Standardize the input data
std_data = scaler.transform(data_reshaped)

In [36]:
std_data

array([[ 0.04601433, -0.34096773,  1.18359575, -1.28821221, -0.69289057,
         0.71168975, -0.84827977, -0.27575966]])

In [40]:
#predicting with our trained Model
predict_data = classifier.predict(std_data)
if (predict_data[0] == 0):
    print('The Person Is not diabetic')
else:
    printnt('The Person is Diabetic')

The Person Is not diabetic
